src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerFormAssetWallet.js

```diff
@@ -1,210 +1,92 @@
-    )
-  )
-}
 
 const propTypes = {
-  metaFormSettings: PropTypes.any,
-  metric: PropTypes.any.isRequired,
-  target: PropTypes.any,
-  setFieldValue: PropTypes.func.isRequired,
-  byAddress: PropTypes.string,
-  assets: PropTypes.array
+  metaFormSettings: PropTypes.any
 }
 
 const TriggerFormAssetWallet = ({
-  data: { allErc20Projects = [], allProjects = [] } = {},
-  target,
   metaFormSettings,
-  metric,
-  assets = [],
-  setFieldValue,
-  byAddress = ''
+  values: { signalType }
 }) => {
   const defaultSignalType = metaFormSettings.signalType
-  const isEthWallet = metric.value === ETH_WALLET_METRIC.value
 
-  const [erc20List, setErc20] = useState(allErc20Projects)
-  const [allList, setAll] = useState(allProjects)
-  const [heldAssets, setHeldAssets] = useState(assets)
+  const { target: defaultAsset } = metaFormSettings
 
-  useEffect(() => {
-    allErc20Projects.length && setErc20(allErc20Projects)
-    allProjects.length && setAll(allProjects)
-    assets.length && setHeldAssets(assets)
-  })
-
-  const canUseMappedErc20 = !!byAddress && assets.length > 0
-  const disabledWalletField = isDisabledWalletAddressField(
-    canUseMappedErc20,
-    target,
-    erc20List
-  )
-  const selectableProjects =
-    canUseMappedErc20 && assets.length > 0 ? assets : allList
+  const isAsset = signalType.value === METRIC_TARGET_ASSETS.value
+  const isWatchlist = signalType.value === METRIC_TARGET_WATCHLIST.value
 
   return (
     <div className={styles.row}>
-      {!isEthWallet && (
+      <div className={styles.Field}>
+        <Label accent='waterloo' className={styles.label}>
+          Type
+        </Label>
+        <FormikSelect
+          name='signalType'
+          disabled={defaultSignalType.isDisabled}
+          defaultValue={defaultSignalType.value.value}
+          placeholder={'Pick signal type'}
+          options={METRIC_TARGET_OPTIONS}
+        />
+      </div>
+
+      {isAsset && (
         <div className={styles.Field}>
-          <Label accent='waterloo' className={styles.label}>
-            Type
-          </Label>
-          <FormikSelect
-            name='signalType'
-            isDisabled={defaultSignalType.isDisabled}
-            defaultValue={defaultSignalType.value.value}
-            placeholder={'Pick signal type'}
-            options={ASSETS_FILTERS}
+          <Label className={styles.label}>&nbsp;</Label>
+          <GetProjects
+            render={({ isLoading, allProjects }) => {
+              return (
+                <FormikSelect
+                  isLoading={isLoading}
+                  name='target'
+                  disabled={defaultAsset.isDisabled}
+                  defaultValue={defaultAsset.value.value}
+                  placeholder='Pick an asset'
+                  required
+                  options={mapToAssets(allProjects, false)}
+                />
+              )
+            }}
           />
         </div>
       )}
-      {isEthWallet && (
+
+      {isWatchlist && (
```

Comment:
if `isAsset` is true, then `isWatchlist` is false and otherwise. Ternary operator could be used here which will result in 1 less condition